openedx/core/djangoapps/user_api/management/commands/populate_retirement_states.py

```diff
@@ -0,0 +1,123 @@
+            raise CommandError('settings.RETIREMENT_STATES ({}) does not contain all required states '
+                               '({})'.format(new_states, REQ_STR))
+
+        # Confirm that the start and end states are in the right places
+        if new_states.index(START_STATE) != 0:
+            raise CommandError('{} must be the first state'.format(START_STATE))
+
+        num_end_states = len(END_STATES)
+
+        if new_states[-num_end_states:] != END_STATES:
+            raise CommandError('The last {} states must be these (in this order): '
+                               '{}'.format(num_end_states, END_STATES))
+
+    def _check_current_users(self):
+        """
+        Check UserRetirementStatus for users currently in progress
+        """
+        if UserRetirementStatus.objects.exclude(current_state__state_name__in=REQUIRED_STATES).exists():
+            raise CommandError(
+                'Users are currently being processed. All users must be in one of these states to run this command: '
+                '{}'.format(REQ_STR)
+            )
+
+    def _delete_old_states_and_create_new(self, new_states):
+        """
+        Wipes the RetirementState table and creates new entries based on new_states
+        - Note that the state_execution_order is incremented by 10 for each entry
+          this should allow manual insert of "in between" states via the Django admin
+          if necessary, without having to manually re-sort all of the states.
+        """
+
+        # Save off old states before
+        current_states = RetirementState.objects.all().values_list('state_name', flat=True)
+
+        # Delete all existing rows, easier than messing with the ordering
+        RetirementState.objects.all().delete()
+
+        # Add new rows, with space in between to manually insert stages via Django admin if necessary
+        curr_sort_order = 1
+        for state in new_states:
+            row = {
+                'state_name': state,
+                'state_execution_order': curr_sort_order,
+                'is_dead_end_state': state in END_STATES,
+                'required': state in REQUIRED_STATES
+            }
+
+            RetirementState.objects.create(**row)
+            curr_sort_order += 10
+
+        # Generate the diff
+        set_current_states = set(current_states)
+        set_new_states = set(new_states)
+
+        states_to_create = set_new_states - set_current_states
+        states_remaining = set_current_states.intersection(set_new_states)
+        states_to_delete = set_current_states - set_new_states
+
+        return states_to_create, states_remaining, states_to_delete
+
+    def handle(self, *args, **options):
+        """
+        Execute the command.
+        """
+        new_states = settings.RETIREMENT_STATES
+        self._validate_new_states(new_states)
+        self._check_current_users()
+        to_create, remaining, to_delete = self._delete_old_states_and_create_new(new_states)
```

Comment:
If the deed was already done, why not call these variables "created, existed, deleted"?

Reply:
Will do